src/hooks/useFilterPossiblyMaliciousPositions.ts

```diff
@@ -0,0 +1,58 @@
+import { Interface } from '@ethersproject/abi'
+import { Token } from '@uniswap/sdk-core'
+import ERC20ABI from 'abis/erc20.json'
+import { Erc20Interface } from 'abis/types/Erc20'
+import { useMultipleContractSingleData } from 'lib/hooks/multicall'
+import { useMemo } from 'react'
+import { PositionDetails } from 'types/position'
+import { hasURL } from 'utils/urlChecks'
+
+import { useDefaultActiveTokens } from './Tokens'
+
+const ERC20Interface = new Interface(ERC20ABI) as Erc20Interface
+
+export function useFilterPossiblyMaliciousPositions(positions: PositionDetails[]): PositionDetails[] {
+  const tokenList = useDefaultActiveTokens()
+
+  const nonListPositionTokenAddresses = useMemo(
+    () =>
+      Array.from(
+        new Set(positions.reduce((acc, position) => acc.concat(position.token0, position.token1), [] as string[]))
+      ).filter((address) => !tokenList[address]),
+    [positions, tokenList]
+  )
+
+  const results = useMultipleContractSingleData(nonListPositionTokenAddresses, ERC20Interface, 'symbol')
```

Comment:
call current chain rpc for info on addresses we don't know already from the list

Reply:
also feel like this should be a separate function here? like there should be a general function that gives you token data without caring if you have it cached or not, which is separate than the purpose of this function

Reply:
Are you imagining something like this?

```
function useTokenContractsConstant(tokens: string[], field: 'name' | 'symbol' | 'decimals' | 'totalSupply') {
  return useMultipleContractSingleData(tokens, ERC20Interface, field, undefined, {
    blocksPerFetch: 10_000,
  })
}
```

Reply:
The only problem is that this is nothing more than a direct use of multi call API which is a hook! And I think at the end of the day this is a pure function, just that it's a hook b/c it uses some hook under the hood. And that's what makes it tricky to cache properly, b/c you can't do that inside `useMemo`.